dataset/dataset_test.go

```diff
@@ -221,11 +236,98 @@ func TestClient_HealthChecker(t *testing.T) {
 	})
 }
 
+func TestClient_GetVersion(t *testing.T) {
+	ctx := context.Background()
+
+	Convey("Given dataset api has a version", t, func() {
+
+		datasetId := "dataset-id"
+		edition := "2023"
+		versionString := "1"
+		versionNumber, _ := strconv.Atoi(versionString)
+		etag := "version-etag"
+
+		version := Version{
+			ID:           "version-id",
+			CollectionID: collectionID,
+			Edition:      edition,
+			Version:      versionNumber,
+			Dimensions: []VersionDimension{
+				{
+					Name:  "geography",
+					ID:    "city",
+					Label: "City",
+				},
+				{
+					Name:  "siblings",
+					ID:    "number_of_siblings_3",
+					Label: "Number Of Siblings (3 Mappings)",
+				},
+			},
+			ReleaseDate:     "today",
+			LowestGeography: "lowest",
+			State:           "published",
+		}
+		httpClient := createHTTPClientMock(MockedHTTPResponse{http.StatusOK, version, map[string]string{"Etag": etag}})
+		datasetClient := newDatasetClient(httpClient)
+
+		Convey("when GetVersion is called", func() {
+			got, err := datasetClient.GetVersion(ctx, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetId, edition, versionString)
+
+			Convey("Then no error is returned", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("And it returns the version", func() {
+				So(got, ShouldResemble, version)
+			})
+			Convey("And the relevant api call has been made", func() {
+				expectedUrl := fmt.Sprintf("/datasets/%s/editions/%s/versions/%s", datasetId, edition, versionString)
+				expectedHeaders := expectedHeaders{
+					FlorenceToken:        userAuthToken,
+					ServiceToken:         serviceAuthToken,
+					CollectionId:         collectionID,
+					DownloadServiceToken: downloadServiceAuthToken,
+				}
+				checkRequestBase(httpClient, http.MethodGet, expectedUrl, expectedHeaders)
+			})
+		})
+
+		Convey("when GetVersionWithHeaders is called", func() {
+			got, h, err := datasetClient.GetVersionWithHeaders(ctx, userAuthToken, serviceAuthToken, downloadServiceAuthToken, collectionID, datasetId, edition, versionString)
```

Comment:
It is your intension that the THEN clauses are tested in isolation with multiple calls to GetVersionWithHeaders ?

Reply:
Not really, but thought it'd read better using separate conveys but I'll join them in one to improve performance 👍